src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,664 @@
+   * @param {DropdownListItem} item - Dropdown selected element.
+   */
+  getA11ySelectedMessage?: (item: DropdownListItem) => string
+
+  /** A function that creates custom accessability message for dropdown item removal.
+   * @param {DropdownListItem} item - Dropdown removed element.
+   */
+  getA11yRemovedMessage?: (item: DropdownListItem) => string
+
+  /** A function that creates custom aria label accessability message for the remove item button.
+   * @param {DropdownListItem} item - The active item to be removed.
+   */
+  getA11yRemoveItemMessage?: (item: DropdownListItem) => string
+
+  /** Array of props for generating dropdown items and selected item labels if multiple selection. */
+  items?: DropdownListItem[]
+
+  /**
+   * Function to be passed to create selected searchQuery from selected item. It will be displayed on selection in the
+   * edit text, for search, or on the button, for non-search. Multiple search will always clear searchQuery on selection.
+   */
+  itemToString?: (item: DropdownListItem) => string
+
+  /** A dropdown can perform a multiple selection. */
+  multiple?: boolean
+
+  /** A string to be displayed when dropdown does not have available items to show. */
+  noResultsMessage?: string
+
+  /**
+   * Called on deletion by backspace when dropdown is a multiple search.
+   *
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All props and proposed value.
+   */
+  onBackspaceDelete?: ComponentEventHandler<DropdownProps>
+
+  /**
+   * Callback for change in dropdown active value(s).
+   * @param {DropdownListItem|DropdownListItem[]} value - Dropdown active value(s).
+   */
+  onDropdownChange?: (value: DropdownListItem | DropdownListItem[]) => any
+
+  /**
+   * Called on clicking the 'X' icon corresponding to an active value.
+   *
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All props and proposed value.
+   */
+  onCloseIconClick?: ComponentEventHandler<DropdownProps>
+
+  /**
+   * Called on key down on the 'X' icon corresponding to an active value.
+   *
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All props and proposed value.
+   */
+  onCloseIconKeyDown?: ComponentEventHandler<DropdownProps>
+
+  /**
+   * Called on container click. Especially useful for multiple search case, when user
+   * can click the container outside the edit text or active values.
+   *
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All props and proposed value.
+   */
+  onContainerClick?: ComponentEventHandler<DropdownProps>
+
+  /**
+   * Called on edit text blur.
+   *
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All props and proposed value.
+   */
+  onInputBlur?: ComponentEventHandler<DropdownProps>
```

Comment:
`input` should be exposed as a slot of `Dropdown`, so that client will be able to pass all the necessary event subscriptions to it directly. This will prevent us from polluting Dropdown's API, as well as allow to reuse already existing Stardust merits:

```jsx
<Dropdown input={{ onFocus: (e) => ..., onKeyDown: (e) => ... }}>
```

Same story generally applies to `label` and `close icon`